### PR TITLE
Fix HTTP logic and add ability to accept extra scopes.

### PR DESCRIPTION
**Pull request by tlwr**
HTTP logic was broken (mb)

If someone wants to give their token access to private repos they can now do so using `extra_scopes` query param (comma separated) (e.g. `&extra_scopes=repo`).

Related to @edds comment on #59

### CODE REVIEW

**Comment by edds** [comment]
> If someone wants to give their token access to private repos they can now do so using extra_scopes query param (comma separated) (e.g. &extra_scopes=repo).

It would be good to put that in the readme.

**Comment by tlwr** [comment]
Just added an extra comments.

File: javascript/preflight.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 function runPreFlightChecks() {
   const isHttp      = window.location.protocol == 'http:',
         isLocalhost = window.location.hostname == 'localhost'
-        isUnsafe    = !(isHttp && isLocalhost);
+        isUnsafe    = !isHttp || !(isHttp && isLocalhost);
```
Comment (robyoung-digital) [inline javascript/preflight.js]: I think this condition is wrong (truth tables below). As far as I understand you're aiming for it being marked as unsafe if you are serving over http and not serving over localhost. ie `isUnsafe = isHttp && !isLocalhost;`

Comment (edds) [reply javascript/preflight.js]: Bah. You're right. It's not working out `isSafe` is working out `isUnsafe`.

Comment (tlwr) [reply javascript/preflight.js]: https://github.com/alphagov/fourth-wall/pull/61

**Pull request merged**